internal/providers/azure/azure.go

```diff
@@ -66,29 +62,40 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
+	var rawConfig []byte
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
+	device, err := execUtil.GetBlockDevices(CDS_FSTYPE_UDF)
 	if err != nil {
-		return types.Config{}, report.Report{}, err
+		return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=UDF: %v", err)
+	} else if len(device) > 0 {
+		for i, dev := range device {
+			rawConfig, err = getRawConfig(f, dev)
+			if len(rawConfig) > 0 {
+				break
+			} else if err != nil && i == len(device) {
+				return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve config: %v", err)
+			}
+		}
 	}
+	return util.ParseConfig(logger, rawConfig)
```

Comment:
We need to keep trying until we either find an Ignition config, or find evidence that the user didn't provide one.